### PR TITLE
Frontend

### PR DESCRIPTION
**Pull request by FabianAlleblas**

### CODE REVIEW

File: frontend/src/Components/IndicatorBars/PlantDetailBar.js [hunk 1]
```diff
@@ -1,28 +1,68 @@
...
+import { useState } from 'react'
 import styled from 'styled-components'
+import { SettingsIcon } from '../Icons'
+import SettingCareIntervalModal from '../Modals/SettingCareIntervalModal'
 
 export default function PlantDetailBar({ daysLeft, weeksLeft }) {
   PlantDetailBar.propTypes = {
     daysLeft: PropTypes.number.isRequired,
     weeksLeft: PropTypes.number.isRequired,
   }
```
Comment (jamarob) [inline frontend/src/Components/IndicatorBars/PlantDetailBar.js:11]:
```suggestion
  PlantDetailBar.propTypes = {
    daysLeft: PropTypes.number.isRequired,
    weeksLeft: PropTypes.number.isRequired,
  }
 
 export default function PlantDetailBar({ daysLeft, weeksLeft }) {
```
No need to redefine PropTypes on every update.

File: frontend/src/Components/IndicatorBars/PlantDetailBar.js [hunk 2]
```diff
@@ -1,28 +1,68 @@
...
 
+  const [isCareSettingShown, setIsCareSettingShown] = useState()
```
Comment (jamarob) [inline frontend/src/Components/IndicatorBars/PlantDetailBar.js:13]: What do you think about just `showCareSetting` for a simpler name?

Maybe init to empty object, or better 
```
{
    water: false,
    fertilizer: false
}
```
to make intention clearer. 

Then later simply `showCareSetting.water &&`

Can both modals be open at the same time, if not, make sure that state is never reached.

File: frontend/src/Components/Modals/SettingCareIntervalModal.js [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+import { ImgDeleteIcon } from '../Icons'
+import PropTypes from 'prop-types'
+
+export default function SettingCareIntervalModal({
+  className,
+  isFertilizer,
+  onClick,
+  onSubmit,
+}) {
+  SettingCareIntervalModal.propTypes = {
```
Comment (jamarob) [inline frontend/src/Components/Modals/SettingCareIntervalModal.js:12]: See comment above

File: frontend/src/Components/Modals/SettingCareIntervalModal.js [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+      <Modal>
+        <ClosingButton onClick={onClick}>
+          <ClosingIcon />
+        </ClosingButton>
+        <Form onSubmit={onSubmit}>
+          <Label>
+            Set your preferred {isFertilizer ? 'fertilizing' : 'watering'}{' '}
+            interval (default: {isFertilizer ? '4 weeks' : '10 days'}):
+            <Input
+              name={isFertilizer ? 'fertilizerInterval' : 'waterIntervat'}
```
Comment (jamarob) [inline frontend/src/Components/Modals/SettingCareIntervalModal.js:30]: Typo

Comment (jamarob) [inline frontend/src/Components/Modals/SettingCareIntervalModal.js:30]: Maybe make these values props to have a really generic component?

**Review by jamarob — COMMENTED** [review]
LGTM 🌱 Left some minor comments.

File: frontend/src/Components/IndicatorBars/PlantDetailBar.js (hunk 1 above)
Comment (FabianAlleblas) [reply frontend/src/Components/IndicatorBars/PlantDetailBar.js:11]: Haha, yeah I have to change it for every file at one point... It's on my list for the last week 😅

File: frontend/src/Components/IndicatorBars/PlantDetailBar.js (hunk 2 above)
Comment (FabianAlleblas) [reply frontend/src/Components/IndicatorBars/PlantDetailBar.js:13]: Good point, at the moment it should never be reached, but I'm going to change it with the init state..

Comment (jamarob) [reply frontend/src/Components/IndicatorBars/PlantDetailBar.js]: I think you can use the function with no modification, say you click the fertilizer button, your state becomes 
```
{
    fertilizer: true
}
```
and `showCareSetting.water` will be falsy and vice versa for the watering button. Prevents the state of both being true, as well.

**Pull request merged**